app/models/gem_dependent.rb

```diff
@@ -0,0 +1,56 @@
+class GemDependent
+  attr_reader :gem_names
+
+  def initialize(gem_names)
+    @gem_information = {}
+    @gem_names = gem_names
+  end
+
+  def fetch_dependencies
+    dependencies = []
+    @gem_names.each { |g| @gem_information[g] = "deps/v1/#{g}" }
+
+    @gem_information.each do |gem_name, cache_key|
+      if memcached_gem_info.keys.include?(cache_key)
```

Comment:
this will cause querying into the hash twice -- why not `if dependency = memcached_gem_info[cache_key] ` ?